receiver/faroreceiver/receiver_test.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package faroreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/faroreceiver"
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+	"go.uber.org/zap"
+)
+
+func TestFaroReceiver_Start(t *testing.T) {
+	testcases := []struct {
+		name               string
+		payload            string
+		expectedStatusCode int
+		expectedTraces     int
+	}{
+		{
+			name:               "empty",
+			payload:            "testdata/empty.json",
+			expectedStatusCode: http.StatusBadRequest,
+			expectedTraces:     0,
+		},
+		{
+			name:               "minimal-traces-only",
```

Comment:
Should have a test for logs (golden test using plogtest.Compare...) and have a golden test for traces using ptracetest.Compare...

Reply:
done, logs, traces, logsandtraces all have golden tests